src/Mailgun/Mailgun.php

```diff
@@ -208,4 +279,28 @@ public function BatchMessage($workingDomain, $autoSend = true)
     {
         return new BatchMessage($this->restClient, $workingDomain, $autoSend);
     }
+
+    /**
+     * @param string $messageUrl
+     *
+     * @return \stdClass
+     */
+    public function getMessageFromUrl($messageUrl = null)
+    {
+        if ($messageUrl === null) {
+          return null;
+        }
+
+        $region = $this->regionForUrl($messageUrl);
+        $regionClient = $this->regionClients[$region];
+        $query = substr(
+            $messageUrl,
+            strspn(
+                $messageUrl,
+                'https://' . $this->regions[$region] . '/' . $regionClient->getApiVersion() . '/'
```

Comment:
Wouldn't `'https://' . $this->regions[$region] . '/' . $regionClient->getApiVersion() . '/'` be equal to the first part of `$messageUrl` at this point?